cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/mazha/CouchClient.java

```diff
@@ -420,8 +419,7 @@ public void putAttachmentStream(String id, String rev, String attachmentName, St
             options.put("att_encoding_info", true);
         }
         options.put("open_revs", jsonHelper.toJson(revisions));
-        return this.getDocument(id, options, new TypeReference<List<OpenRevision>>() {
-        });
+        return this.getDocument(id, options,new CouchClientTypeReference<List<OpenRevision>>());
```

Comment:
This changed slightly in a later commit, it now uses the `JavaType` class from Jackson, (`TypeReference`s don't work as static class, not sure why.) The rest is still valid, I'll add the fields as required.